relayer/provider/provider.go

```diff
@@ -44,6 +44,23 @@ type LatestBlock struct {
 	Time   time.Time
 }
 
+type IBCHeader interface {
+	IBCHeader()
+}
```

Comment:
> By convention, one-method interfaces are named by the method name plus an -er suffix or similar modification to construct an agent noun: Reader, Writer, Formatter, CloseNotifier etc.

https://go.dev/doc/effective_go#interface-names

I don't really want to name this `IBCHeaderer` but `IBCHeader` doesn't follow the convention either.

Reply:
maybe `IBCHeaderIndicator` ?